### DISCUSSION TITLE
객체끼리 메시지를 주고 받는 부분에서 이렇게 하면 될까요?

### DISCUSSION THREAD

**Discussion by Joseph-Cha** [Joseph-Cha/JustFight_Client#18] [Q&A] [answered]
### 기능별 요구사항 리스트 및 설계 
1. 앱 실행 후 인게임에 들어가기 전까지
     1. 요구사항
        - 앱 실행 시 타이틀 화면 등장
        - 타이틀 아래에서 (UI)"Touch To Start"가 깜빡 거림
        - (UI)"Touch To Start"를 터치하면 메인 씬으로 화면 전환 
        - 화면 전환할 때 Fade In & Out
    2. 설계
        - `유저`가 화면을 터치를 해서 `Button`에게 `화면 터치` 메시지를 보낸다.
        - `Button`이 `SceneLoader`에게 `OnClick()` 메시지를 보낸다.
        - `SceneLoader`가 `자신`에게 `OnStart()` 메시지를 보낸다.

### 기능별 요구사항 리스트 및 설계 
1. 앱 실행 후 인게임에 들어가기 전까지
    1. (UI)"Touch To Start"를 터치하면 메인 씬으로 화면 전환
        - `유저` ⇒ `Button` : `화면 터치`
        - `Button` ⇒ `SceneLoader` : `OnClick() : void`
        - `SceneLoader` ⇒ `SceneLoader` : `OnStart() : void`
    2. 타이틀 아래에서 (UI)"Touch To Start"가 깜빡 거림
        - `Animator` ⇒ `Touch To Start(UI)` : `Play() : void`
    3. 화면 전환할 때 Fade In & Out
        - `SceneLoader` ⇒ `Animator` : `SetTrigger(string) : void`
        - `Animator` ⇒ `BlackView(UI)` : `Play() : void`

둘중에 어떤게 더 가독성이 좋은가요?

**Comment by jongfeel** [comment, marked as answer]
저는 둘 다 괜찮습니다. 가독성이 중요하지는 않은 거 같아요. 뭘 하고 싶어하는지가 중요하니까요.

첫 번째가 요구사항과 설계를 별도로 적은거 같고
두 번째가 요구사항과 설계를 각 단계별로 적어 둔 것 같습니다.

첫 번째 형태로 요구사항은 그대로 두고
설계 부분을 화면 설계와 객체 설계로 혼합해서 적어보면 좋을 것 같아요.

그림을 그리면 쉬운데 글로 적어도 된다고 했으니 글로 적어 봅니다.

### Screen design

legend
1 depth = object and name
2 depth = properties and descriptions

- Title
  - Text: "just fight"
  - Position: center
- Button
  - Text: "touch to start"
  - Position: below the title
  - Effect: blink every 0.5 second
  - Click: load next scene
    - Effect: fade in/out during 2 seconds

화면 설계의 포인트는 한 화면의 스냅샷과 더불어 그 화면을 설명하는 내용이 있으면 됩니다.

### Object collaboration design

legend
italic: message
bold: object

_start_ -> **Title**
_start_ -> **Button** -> _load scene_ -> **SceneLoader** -> _play_ -> **Animator**

객체간의 연결 관계가 있고 메시지를 전달받아 실행하는 관계라면 위와 같이 생각해 볼 수 있습니다.

**Reply by jongfeel** [reply]
제 기준으로 이정도 설계 하면 코드 나올 것 같습니다.

**Reply by Joseph-Cha** [reply]
가이드 감사합니다 ㅎㅎ 참고해서 커밋할게요!
